consensus/polybft/burnt_fees_manager.go

```diff
@@ -0,0 +1,140 @@
+package polybft
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+const (
+	// defaultBurntFeesWithdrawalOffset is the frequency at which withdrawals are sent to the rootchain (in blocks)
+	defaultBurntFeesWithdrawalOffset = uint64(900)
+)
+
+type BurntFeesManager interface {
+	PostBlock(req *PostBlockRequest) error
+}
+
+// dummyBurntFeesManager is the dummy implementation of BurntFeesManager
+type dummyBurntFeesManager struct {
+}
+
+func (m *dummyBurntFeesManager) PostBlock(req *PostBlockRequest) error {
+	return nil
+}
+
+// burntFeesManager implements BurntFeesManager interface by implementing a logic that
+// sends the withdrawal transaction to the EIP1559Burnt contract once an epoch.
+// This is needed to automatically sync burnt fees between root chain and child chain.
+type burntFeesManager struct {
+	// key is the identity of the node submitting a checkpoint
+	key ethgo.Key
+
+	// txRelayer is the abstraction on the child chain interaction logic.
+	txRelayer txrelayer.TxRelayer
+
+	// calculateBurnContract returns burnt contract address based on the given block
+	calculateBurnContract func(block uint64) (types.Address, error)
+
+	// forks is needed to check if london hardfork is enabled for the given block
+	forks *chain.Forks
+
+	// checkpointsOffset represents offset between withdrawal blocks (applicable only for non-epoch ending blocks)
+	withdrawalOffset uint64
```

Comment:
Do we need the offset if we are going to do this once per epoch?

Reply:
Just in case if something went wrong during the epoch block run so it could be started again based on the offset.

Reply:
Correct, we just need to add it to the pool basically. But the question is what should be the tx type and who is going to pay for it?

Reply:
Yeah. Maybe it might make sense to even have it as a state transaction, since it is a form of `automatic` burning of child tokens.